Formula/i686-elf-gcc.rb

```diff
@@ -47,6 +49,9 @@ def install
       # FSF-related man pages may conflict with native gcc
       (share/"man/man7").rmtree
     end
+
+    # Dir file will conflict with others, is regenerated on install
+    rm_f "#{info}/#{target}/dir"
```

Comment:
What is the `dir` file, and how is it regenerated on install?

Reply:
It's the "table of contents" for a GNU info directory, and is automatically generated by running `install-info` when `brew link`ing every info file: https://github.com/Homebrew/brew/blob/dc2c0ec8cf742b9499b7a765d3a4b44a2e5c3367/Library/Homebrew/keg.rb#L630-L634

Except I just realized that if only one of the arch's utility formulae is installed (e.g. `i686-elf-binutils` but not `i686-elf-gcc`), then Homebrew would just link the entire `info/i686-elf` directory into `/usr/local/share/info`, and then we''d have no `dir` in that directory at all.

This requires more thinking.

Reply:
OK, based on reading and some experimentation, this seems to be the state of Homebrew-and-GNU-Info:

1. Homebrew avoids bottling `**/info/dir` files, but _does_ bottle `**/info/<subdir>/dir`. (This explains why these bottles are the only ones I've found that contain `dir` files.)
2. Homebrew regenerates `dir` files when `brew link`ing individual `*.info` files, regardless of their path.
3. Regenerated `dir` files are stored in the same directory as the `.info` files they index.
4. The `info` command only consults the _root_ index, i.e. running `info` only displays the contents of `<prefix>/share/info/dir`.
5. Info files in subdirectories of the Info root dir can be directly access via their "partial paths" (e.g. `info i686-elf/as`).

It therefore seems that the Right Thing is to remove `dir` across _all_ the formulae, including `arm-linux-gnueabihf-binutils`, since they won't be used under normal circumstances anyway, and will cause conflicts between two formulae for the same arch target. (We _could_ add code to automate this in Homebrew itself, but 6 out of several thousand formulae doesn't seem enough to justify the effort.)

Reply:
> 1. Homebrew avoids bottling `**/info/dir` files, but _does_ bottle `**/info/<subdir>/dir`. (This explains why these bottles are the only ones I've found that contain `dir` files.)

I take that back. Homebrew actually _deletes_ `dir` files during `brew install --build-bottle`:
```
...
$ brew install --build-bottle grep
...
==> Cleaning
==> Fixing /usr/local/Cellar/grep/3.6/bin/gegrep permissions from 755 to 555
==> Fixing /usr/local/Cellar/grep/3.6/bin/gfgrep permissions from 755 to 555
==> Fixing /usr/local/Cellar/grep/3.6/bin/ggrep permissions from 755 to 555
rm /usr/local/Cellar/grep/3.6/share/info/dir
==> Finishing up
ln -s ../Cellar/grep/3.6/bin/gegrep gegrep
ln -s ../Cellar/grep/3.6/bin/gfgrep gfgrep
ln -s ../Cellar/grep/3.6/bin/ggrep ggrep
ln -s ../../Cellar/grep/3.6/share/info/grep.info grep.info
info /usr/local/share/info/grep.info
ln -s ../../../Cellar/grep/3.6/share/man/man1/gegrep.1 gegrep.1
ln -s ../../../Cellar/grep/3.6/share/man/man1/gfgrep.1 gfgrep.1
ln -s ../../../Cellar/grep/3.6/share/man/man1/ggrep.1 ggrep.1
==> Not running 'post_install' as we're building a bottle
You can run it manually using:
  brew postinstall grep
...
```
>  It therefore seems that the Right Thing...

...is to fix the cleanup to _recursively_ remove all `dir` files under `info/`.

Reply:
I'll push one last time to remove all these `rm_f ... dir`s, after https://github.com/Homebrew/brew/pull/11597 is merged.